packages/ketcher-core/src/application/editor/modes/SequenceMode.ts

```diff
@@ -154,53 +154,125 @@ export class SequenceMode extends BaseMode {
+          editor,
+          labeledNucleoelement.phosphateLabel,
+        );
+      }
 
       const currentNode = SequenceRenderer.getNodeByPointer(nodeIndexOverall);
 
-      // Update monomerItem objects
-      modelChanges.merge(
-        editor.drawingEntitiesManager.modifyMonomerItem(
-          currentNode.sugar,
-          sugarMonomerItem as MonomerItemType,
-        ),
-      );
-      modelChanges.merge(
-        editor.drawingEntitiesManager.modifyMonomerItem(
-          currentNode.rnaBase,
-          baseMonomerItem as MonomerItemType,
-        ),
-      );
-      modelChanges.merge(
-        editor.drawingEntitiesManager.modifyMonomerItem(
-          currentNode.phosphate,
-          phosphateMonomerItem as MonomerItemType,
-        ),
-      );
+      // Update Sugar monomerItem object
+      if (currentNode.sugar && sugarMonomerItem) {
+        modelChanges.merge(
+          editor.drawingEntitiesManager.modifyMonomerItem(
+            currentNode.sugar,
+            sugarMonomerItem,
+          ),
+        );
+      }
+      // Update Base monomerItem object
+      if (currentNode.rnaBase && baseMonomerItem) {
+        modelChanges.merge(
+          editor.drawingEntitiesManager.modifyMonomerItem(
+            currentNode.rnaBase,
+            baseMonomerItem,
+          ),
+        );
+      }
+
+      // Update monomerItem object or add Phosphate
+      if (phosphateMonomerItem) {
+        // Update Phosphate monomerItem object for Nucleotide
+        if (currentNode instanceof Nucleotide) {
+          modelChanges.merge(
+            editor.drawingEntitiesManager.modifyMonomerItem(
+              currentNode.phosphate,
+              phosphateMonomerItem,
+            ),
+          );
+          // Add Phosphate to Nucleoside
+        } else if (currentNode instanceof Nucleoside) {
+          const sugarR2 = currentNode.sugar.attachmentPointsToBonds.R2;
+          const nextMonomerInSameChain = sugarR2?.secondMonomer;
+          const newPhosphateCreationCommand =
+            editor.drawingEntitiesManager.addMonomer(
+              phosphateMonomerItem,
+              new Vec2(0, 0),
+            );
+          const newPhosphate =
+            newPhosphateCreationCommand.operations[0].monomer;
+
+          if (!newPhosphate) continue;
+
+          modelChanges.merge(newPhosphateCreationCommand);
+          // Remove existing bond connection between Nucleoside Sugar and next node in case of any
+          if (sugarR2) {
+            modelChanges.merge(
+              editor.drawingEntitiesManager.deletePolymerBond(sugarR2),
+            );
+          }
+          // Connect Nucleoside Sugar with new Phosphate
+          modelChanges.merge(
+            editor.drawingEntitiesManager.createPolymerBond(
+              currentNode.sugar,
+              newPhosphate,
+              'R2',
+              'R1',
+            ),
+          );
+          if (nextMonomerInSameChain) {
+            if (nextMonomerInSameChain) {
```

Comment:
I believe we can try to unify this part with bondNodesThroughNewPhosphate method. So my proposal is to work with nodes here instead of separate monomers as much as possible because it is higher level of abstraction.

Reply:
Yes, we can use `bondNodesThroughNewPhosphate` here. But some updates of `bondNodesThroughNewPhosphate` needed.

Reply:
Updated!